tests/ProxyManagerTest/Functional/RemoteObjectFunctionalTest.php

```diff
@@ -33,12 +36,15 @@
 final class RemoteObjectFunctionalTest extends TestCase
 {
     /**
-     * @param mixed   $expectedValue
+     * @param mixed $expectedValue
+     * @param string $method
      * @param mixed[] $params
+     * @return XmlRpcAdapter
      */
-    protected function getXmlRpcAdapter($expectedValue, string $method, array $params) : XmlRpcAdapter
+    protected function getXmlRpcAdapter($expectedValue, string $method, array $params): XmlRpcAdapter
     {
-        $client = $this->getMockBuilder(Client::class)->setMethods(['call'])->getMock();
+        /** @var Client|MockObject $client */
+        $client = $this->getMockBuilder(Client::class)->getMock();
```

Comment:
Yeah, I realised that you have done that because the class was not defined at all. 

Now I've done as you suggested, much better this way indeed. I thought of copying the class but didn't know how to register it as the same namespace of the original library.